splunk-otel-android/src/main/java/com/splunk/rum/StartTypeAwareSpanStorage.java

```diff
@@ -0,0 +1,136 @@
+
+import static com.splunk.rum.SplunkRum.LOG_TAG;
+
+import android.util.Log;
+import io.opentelemetry.android.instrumentation.activity.VisibleScreenTracker;
+import java.io.File;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+/***
+ * Store span files to /span/background/@uniqueId/ for spans created when the app started in the background.
+ * If the app is brought to foreground and the same session ID still in use, the background spans are moved to /span for eventual sending.
+ * If the app still in the background until process-kill, the background span files will eventually be deleted by the @DeviceSpanStorageLimiter.
+ */
+public class StartTypeAwareSpanStorage implements SpanStorage {
+
+    private final VisibleScreenTracker visibleScreenTracker;
+    private final FileUtils fileUtils;
+    private final String uniqueId;
+    private final File rootDir;
+    private final File spanDir;
+
+    public StartTypeAwareSpanStorage(
+            VisibleScreenTracker visibleScreenTracker, FileUtils fileUtils, File rootDir) {
+        this.visibleScreenTracker = visibleScreenTracker;
+        this.fileUtils = fileUtils;
+        this.rootDir = rootDir;
+        this.spanDir = fileUtils.getSpansDirectory(rootDir);
+        this.uniqueId = UUID.randomUUID().toString();
+        // if new id then background span directory with old ids can be deleted
+        cleanUpUnsentBackgroundSpan();
+    }
+
+    @Override
+    public long getTotalFileSizeInBytes() {
+        return fileUtils.getTotalFileSizeInBytesRecursively(spanDir);
+    }
+
+    @Override
+    public Stream<File> getAllSpanFiles() {
+        return fileUtils.listFilesRecursively(spanDir);
+    }
+
+    @Override
+    public Stream<File> getPendingFiles() {
+        if (isAppForeground()) {
+            moveBackgroundSpanToPendingSpan();
+        }
+        return fileUtils.listSpanFiles(spanDir);
+    }
+
+    private boolean isAppForeground() {
+        return (visibleScreenTracker.getCurrentlyVisibleScreen() != null
+                        && !visibleScreenTracker.getCurrentlyVisibleScreen().equals("unknown"))
+                || visibleScreenTracker.getPreviouslyVisibleScreen() != null;
+    }
+
+    private void moveBackgroundSpanToPendingSpan() {
+        fileUtils
+                .listSpanFiles(getCurrentSessionBackgroundFile())
+                .forEach(
+                        file -> {
+                            File destinationFile = new File(spanDir, file.getName());
+                            boolean isMoved = file.renameTo(destinationFile);
+                            Log.d(
+                                    LOG_TAG,
+                                    "Moved background span "
+                                            + file.getPath()
+                                            + " success ? "
+                                            + isMoved
+                                            + " for eventual send");
+                        });
```

Comment:
I think we should remove the subdirectory here as well.
```suggestion
        File subdir = getCurrentSessionBackgroundFile();
        fileUtils.listSpanFiles(subdir)
                .forEach(
                        file -> {
                            File destinationFile = new File(spanDir, file.getName());
                            boolean isMoved = file.renameTo(destinationFile);
                            Log.d(
                                    LOG_TAG,
                                    "Moved background span "
                                            + file.getPath()
                                            + " success ? "
                                            + isMoved
                                            + " for eventual send");
                        });
        fileUtils.safeDelete(subdir);
```